### ISSUE TITLE
Auto correct with ternary operator (?:) and round brackets creates invalid code due to missing space

### ISSUE THREAD

**Issue by stillhart** [rubocop/rubocop#10329]
I like to use and misuse the ternary operator a lot. I run into an issue while running `rubocop --auto-correct` on my ternary operator code. Originally it happened with jbuilder in rails but it can be easily reproduced in a minimal example. Rubocop forgets to add a space when removing round brackets. 

Here is some context:
```ruby
# A method with some args
def asdf(*args,**opts)
end
# Or how I was originally using rails jbuilder
json.asdf some_ternary_code
```

## Example
Now when I run `rubocop --auto-correct` with following lines I get the underneath actual behaviour

Failing example:
```ruby
json.asdf (foo || bar) ? 1 : 2
```

Successful example:
```ruby
json.asdf((foo || bar) ? 1 : 2)
```

--------

## Expected behavior

```ruby
json.asdf foo || bar ? 1 : 2
```
```ruby
json.asdf(foo || bar ? 1 : 2)
```

## Actual behavior

Rubocop creates invalid code due to the missing space
```ruby
json.asdffoo || bar ? 1 : 2
#       /\
#       ^^ space missing, this can fail potentially and is not desirable
```

The other example works as desired. 
```ruby
json.asdf(foo || bar ? 1 : 2)
```

## Steps to reproduce the problem

Install rubocop
```
gem install rubocop -v 1.24.0
```

Create a file with the following content:
```ruby
# tenary.rb
json.asdf (foo || bar) ? 1 : 2
json.asdf((foo || bar) ? 1 : 2)
```

Run rubocop in auto correct mode on the file
`rubocop --auto-correct tenary.rb`

```
tenary.rb:2:10: W: [Corrected] Lint/ParenthesesAsGroupedExpression: (...) interpreted as grouped expression.
json.asdf (foo || bar) ? 1 : 2
         ^
tenary.rb:2:11: C: [Corrected] Style/TernaryParentheses: Omit parentheses for ternary conditions.
json.asdf (foo || bar) ? 1 : 2
          ^^^^^^^^^^^^^^^^^^^^
```

## RuboCop version

```
bundle exec rubocop -V
1.24.0 (using Parser 3.0.3.2, rubocop-ast 1.15.1, running on ruby 3.1.0 x86_64-linux)
```

## Note

I never thought that this code could be vague but it easily could be interpreted as a method argument instead. Ruby seems to be doing the right thing though.
```ruby
json.asdf (foo || bar) ? 1 : 2
json.asdf(foo || bar) ? 1 : 2
```
